python_backend/langchain_pipeline/retriever/vector_db.py

```
@@ -28,24 +28,80 @@
+        # 허용된 디렉토리 목록
+        allowed_dirs = [
+            "python_backend/langchain_pipeline/data",
+            "data/faiss_index",
+            "langchain_pipeline/data"
+        ]
+
+        for allowed_dir in allowed_dirs:
+            allowed_path = project_root / allowed_dir
+            try:
+                abs_path.relative_to(allowed_path.resolve())
+                return True
```

Ruff (0.13.1)

Consider moving this statement to an `else` block

(TRY300)